### PR TITLE
informant payload fix for informant change

### PR DESCRIPTION
**Pull request by Nil20**

### CODE REVIEW

File: packages/client/src/views/RegisterForm/RegisterForm.tsx [hunk 1]
```diff
@@ -686,6 +686,10 @@ class RegisterFormView extends React.Component<FullProps, State> {
         informant = 'father'
       }
 
+      // informant needs to be reset as the relationship changed
+      // see https://github.com/opencrvs/opencrvs-core/issues/5866
+      delete declaration.data.informant._fhirIDPatient
+
```
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/RegisterForm.tsx:692]: We should not directly update the declaration data as it's in redux. Perhaps we can do something like this:
```
const { _fhirIDPatient, ...informantWithoutID } = declaration.data.informant
this.props.modifyDeclaration({
  ...declaration,
  data: {
    ...
    ...
    informant: {
      ...informantWithoutID
    }
  }
```

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/client/src/views/RegisterForm/RegisterForm.tsx (hunk 1 above)
Comment (Nil20) [reply packages/client/src/views/RegisterForm/RegisterForm.tsx:692]: what would be its advantage?

Comment (Zangetsu101) [reply packages/client/src/views/RegisterForm/RegisterForm.tsx]: in this case I guess it doesn't give us anything as the "_fhirIDPatient" isn't being used anywhere in the client but the reasoning behind this approach would be the "immutability pattern". Using "delete" doesn't create a new object so the redux store wouldn't have any idea that the object changed

File: packages/client/src/views/RegisterForm/RegisterForm.tsx [hunk 2]
```diff
@@ -686,13 +686,17 @@ class RegisterFormView extends React.Component<FullProps, State> {
         informant = 'father'
       }
 
+      // informant needs to be reset as the relationship changed
+      // see https://github.com/opencrvs/opencrvs-core/issues/5866
       if (informant) {
+        const { _fhirIDPatient, ...informantWithoutFhirID } =
+          declaration.data[informant]
```
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/RegisterForm.tsx:693]:
```suggestion
          declaration.data.informant
```
The informant & father/mother details need to be handled separately

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**